tools/gbench/report.py

```diff
@@ -266,11 +274,75 @@ def get_color(res):
     return output_strs
 
 
-###############################################################################
-# Unit tests
+def get_json_difference_report(
```

Comment:
This is bad.
Now there are two different functions that *should* only differ with the output format.
Meaning, every change will need to happen in two places.
And extra attention will need to be paid to prevent them from diverging.

Instead, `get_json_difference_report()` should be the only one,
and `generate_difference_report()` should be fed the output of `get_json_difference_report()`,
and it should just pretty-print it.

Reply:
Alright, I think I'm on my way finishing this change. Please see 9021718. `generate_difference_report` is now a consumer of data fed by `get_json_difference_report`.

One more thing I've yet to be done is to re-implement the unit-tests to complement this change.

Reply:
One more thing I would like to note and which is related to your comment in #740 is that `print_utest` now became much less involved and looks more natural to what is supposed to actually do.